src/channel.ts

```diff
@@ -1616,19 +1616,19 @@ export class Channel<
         }
         break;
       case 'reaction.new':
-        if (event.reaction) {
-          channelState.addReaction(event.reaction, event.message);
+        if (event.message && event.reaction) {
+          event.message = channelState.addReaction(event.reaction, event.message);
         }
         break;
       case 'reaction.deleted':
         if (event.reaction) {
-          channelState.removeReaction(event.reaction, event.message);
+          event.message = channelState.removeReaction(event.reaction, event.message);
         }
         break;
       case 'reaction.updated':
         if (event.reaction) {
           // assuming reaction.updated is only called if enforce_unique is true
-          channelState.addReaction(event.reaction, event.message, true);
+          event.message = channelState.addReaction(event.reaction, event.message, true);
```

Comment:
why is this mutation needed ?

Reply:
We want subscribers of reaction events to receive correct `own_reactions` field. We synchronise contents of this fields on the client side, so in order to propagate updated field to subscribers, we need to update it. Updating message object  implicitly in channels state didn't seem right on the previous code review (see @vishalnarkhede comment above)

Reply:
Got it, LGTM, thanks